.github/workflows/code_coverage.yml

```diff
@@ -6,51 +6,36 @@ jobs:
   Codecov:
     name: Code Coverage
     runs-on: ubuntu-latest
-    env:
-      RUSTFLAGS: "-Cinstrument-coverage"
-      RUSTDOCFLAGS: "-Cinstrument-coverage"
-      LLVM_PROFILE_FILE: "./target/coverage/%p-%m.profraw"
-
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Install lcov tools
-        run: sudo apt-get install lcov -y
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-            toolchain: stable
-            override: true
-            profile: minimal
-            components: llvm-tools-preview
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.2.1
-      - name: Install grcov
-        run: if [[ ! -e ~/.cargo/bin/grcov ]]; then cargo install grcov; fi
-      - name: Build simulator image
-        run: docker build -t hwi/ledger_emulator ./ci -f ci/Dockerfile.ledger
-      - name: Run simulator image
-        run: docker run --name simulator --network=host hwi/ledger_emulator &
-      - name: Install Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.9'
-      - name: Install python dependencies
-        run: pip install hwi==2.1.1 protobuf==3.20.1
-      - name: Test
-        run: cargo test --all-features
-      - name: Make coverage directory
-        run: mkdir coverage
-      - name: Run grcov
-        run: grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --keep-only '**/crates/**' --ignore '**/tests/**' --ignore '**/examples/**' -o ./coverage/lcov.info
-      - name: Generate HTML coverage report
-        run: genhtml -o coverage-report.html --ignore-errors source ./coverage/lcov.info
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Code Coverage
+      - name: Code Coverage
+        run: nix build -L .#ci.codeCoverage
```

Comment:
**suggestion:** Since ccov uses a completely different target and can't really reuse artifacts with normal builds, it benefits from running as a separate job, so it can run in parallel.